cmake/modules/boards.cmake

```
@@ -25,6 +25,11 @@
 # - ARCH_DIR:       Arch dir for extracted from selected board
 # - BOARD_ROOT:     BOARD_ROOT with ZEPHYR_BASE appended
 #
+# If BOARD_EXTENSIONS is enabled, the following variables will also be defined:
```

how is this enabled/disabled? This seems to be always enabled here.

---

The feature is meant to be enabled by default, same as sample overlays. It is basically a safeguard in case one inherits a misbehaving module.

---

`cmake -DBOARD_EXTENSIONS=off ...`

---

well, if it is enabled by default, then saying ` If BOARD_EXTENSIONS is enabled` is misleading. 

if not disabled sounds better, and then probably add a hint how to disable it.

---

improved description, details on how to disable the option are provided in the board docs.